lib/api/dsl/methods.js

```diff
@@ -551,11 +665,89 @@ module.exports = methods = {
 
   /**
    * Return true only if the point in field is in a specific distance from a geo point
+   * @param {String} roomId
+   * @param {String} collection
+   * @param {Object} filter given by user on subscribe
+   * @param {Boolean} not if not is true, invert the boolean result
+   * @return {Promise} the formatted filter that need to be added to the room
    */
-  geoDistance: function () {
-    var deferred = q.defer();
+  geoDistance: function (roomId, collection, filter, not) {
+    var
+      curriedFunctionName,
+      deferred = q.defer(),
+      fieldName,
+      formattedFilters = {},
+      geoFilter,
+      point,
+      lat,
+      lon,
+      distance,
+      result;
+
+    if (_.isEmpty(filter)) {
+      deferred.reject(new BadRequestError('Missing filter'));
+      return deferred.promise;
+    }
+
+    fieldName = Object.keys(filter)[0];
+    if (fieldName === 'distance') {
+      fieldName = Object.keys(filter)[1];
+    }
+
+    geoFilter = filter[fieldName];
```

Comment:
If the filter contains only a ``distance`` member and nothing else, then ``fieldName`` is undefined. Since this argument comes from users, this test must be tested.